### PR TITLE
[meshery] fix bug docker couldnt find swagger.yaml

### LINKED ISSUE #3642: [MESHERY] swagger.yaml not found when running meshery server in docker
#### Description

Swagger.yaml is a result of running the Open API for documenting our REST APIs on Meshery Server. It could be produced using make commands available in makefile. We can see them in development, i.e. by running them using `make run-local`. But when running its image using Docker `mesheryctl system start`, the documentation route could not found `swagger.yaml` hence making us cannot view the APIs.

#### Expected Behavior

We can see the documentation of REST APIs route filled with documented API

#### Environment

- Host OS: Linux 
- Meshery Server Version: stable
- Meshery Client Version: stable
- Platform: Kubernetes

---

**Contributor Resources:**
- (Contributing Docs](https://docs.meshery.io/project/contributing)

### PR DESCRIPTION
**Pull request by bariqhibat**
Signed-off-by: Bariq <bariqhibat@gmail.com>

**Description**

This PR fixes #3642

**Notes for Reviewers**

Since docker-compose.yaml uses the deployed docker image of Meshery, you will see that the /docs is still not working. 
So, you need to change `docker-compose.yaml` to build the docker image manually and uses it for running the cluster. 
Change `docker-compose.yaml` line 4:
```
    image: layer5/meshery:stable-latest
```
to 
```
    build: 
      context: .
      dockerfile: Dockerfile
```

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -53,4 +54,4 @@ RUN curl -Lo nighthawk_output_transform-ubuntu-amd64-v1.0.1.tar.gz https://githu
 RUN mkdir -p /home/appuser/.meshery/config; chown -R appuser /home/appuser/
 USER appuser
 WORKDIR /app/cmd
-CMD ./meshery
+ENTRYPOINT ./meshery
```
Comment (leecalcote) [inline Dockerfile:57]: @bariqhibat why the change from CMD to ENTRYPOINT?

Comment (bariqhibat) [reply Dockerfile:57]: no particular reason really, i just followed some of the solution that i found online. should i change it back?

Comment (leecalcote) [reply Dockerfile:57]: Yes, please.

Comment (bariqhibat) [reply Dockerfile]: sure, just changed it!

**Comment by leecalcote** [comment]
Thanks.

I _think_ exposing API docs in a deployment is desirable, but I'm unconvinced at the moment. Will you say more about how you think this will be used?

**Comment by bariqhibat** [comment]
> Thanks.
> 
> I _think_ exposing API docs in a deployment is desirable, but I'm unconvinced at the moment. Will you say more about how you think this will be used?

Automated REST API docs in deployment makes a good replacement for our manual creation of [REST API reference](https://docs.meshery.io/reference/rest-apis). The pros of the automated REST API docs is that, it adds a better clarity on the types of API it is (the tags, whether it is Performance, etc), how your request should be arranged and how the response is going to be. Another thing, it normalizes how the docs should be, hence indifferent quality of one to another. However, as you can presume, the cons would be less flexibility and editability on the docs from our side

**Comment by leecalcote** [comment]
@bariqhibat I've build and run this container. When I visit http://localhost:9081/docs, I get:

Screen Shot 2021-07-27 at 7 08 55 PM


What am I doing wrong?

**Comment by bariqhibat** [comment]
> @bariqhibat I've build and run this container. When I visit http://localhost:9081/docs, I get:
> 
> Screen Shot 2021-07-27 at 7 08 55 PM
> 
> What am I doing wrong?

_Shoot_, sorry Lee, it seems like my previous tries has been confused by docker build cache, I've pushed a new commit where I do `docker build prune` prior to testing it. Could you try pruning the image and build it again? Thanks!

**Comment by bariqhibat** [comment]
Hey @leecalcote, is the fix still not working?

**Comment by piyushsingariya** [comment]
@bariqhibat can you show if it's working for you?

**Comment by bariqhibat** [comment]
> @bariqhibat can you show if it's working for you?

Yep, I checked it

**Comment by leecalcote** [comment]
Hopefully, this aids a fair bit as it's more complete documentation than what is currently published on Meshery Docs.
